packages/flame/lib/src/components/text_box_component.dart

```diff
@@ -93,9 +112,13 @@ class TextBoxComponent<T extends TextRenderer> extends PositionComponent {
 
   bool get finished => _lifeTime > totalCharTime + _boxConfig.dismissDelay;
 
+  int get _actualTextLength {
+    return _lines.map((e) => e.length).fold(0, (p, c) => p + c);
+  }
+
   int get currentChar => _boxConfig.timePerChar == 0.0
-      ? _text.length
-      : math.min(_lifeTime ~/ _boxConfig.timePerChar, _text.length);
+      ? _actualTextLength
+      : math.min(_lifeTime ~/ _boxConfig.timePerChar, _actualTextLength);
```

Comment:
it is possible that the length of the line could be different than the original text length because we might remove spaces that became line breaks. this is technically more accurate, though it was hardly the issue I so desperately sought and probably irrelevant in pretty much all cases.